Created food module in modules.src

finleylau opened the pull request:
Added to modules.src __init__.py
Added tests in test_food.py in modules.tests
Added API key in config.py

finleylau commented:
Some examples of testing queries:
Is Giordano's pizza in Chicago good?
What are some restaurants in San Francisco?
Chipotle in New York

finleylau commented:
@swapagarwal are you able to train the wit.ai so I can see if the tests pass?  Thanks!

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+        
+        location = query['location_suggestions'][0]
+
+        search_url = 'https://developers.zomato.com/api/v2.1/search?entity_id={eid}&entity_type={etype}&lat={lat}&lon={lon}&sort=rating&order=desc'.format(eid=location['entity_id'],etype=location['entity_type'],lat=location['latitude'],lon=location['longitude'])
```
Let's split the line please, hard to follow :)

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+        requested_location = entities['restaurant_location'][0]['value']
+
+        header = {'user-key': ZOMATO_API_KEY, 'Accept': 'application/json'}
+        query = requests.get('https://developers.zomato.com/api/v2.1/locations?query=' + requested_location, headers=header).json()
```
Are we sure that JSON conversion wouldn't fail? What happens when the connection is dropped or malformed response received. Good to have it handled :)

horrormyth replied:
lets start with renaming the query as the response that is more readable ,and then we shall do ‘‘‘ if response.ok'''

horrormyth replied:
lets start with renaming the query as the response that is more readable , and then we shall do 
```
if response.ok:
    content = response.json()
    try:
        location=content['location_suggestions'][0]
    except (KeyError,IndexError):
        #do what you gotta do.
    #do normal thing
```
This is still hacky way but at least we have something to catch if fails.An I hope this will make it clear :)

finleylau replied:
updated with your changes, is that what you were looking for?

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+            type = entities['cuisine'][0]['value']
+            search_url += '&cuisines={type}'.format(type=type)
+
+        query = requests.get(search_url,headers=header).json()['restaurants']
```
Same as before, this one seems prone to failure.

horrormyth commented on .vs/ProjectSettings.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "CurrentProjectSetting": null
+}
```
Let's start with adding the blank line at the end of each file :)  (POSIX standard)

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+        search_url = 'https://developers.zomato.com/api/v2.1/search?entity_id={eid}&entity_type={etype}&lat={lat}&lon={lon}&sort=rating&order=desc'.format(eid=location['entity_id'],etype=location['entity_type'],lat=location['latitude'],lon=location['longitude'])
+        
+        name = ''
+        type = ''
```
Let's avoid using inbuilt keywords

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+            search_url += '&q={name}'.format(name=name)
+
+        if 'cuisine' in entities:
+            type = entities['cuisine'][0]['value']
```
Same as before lets avoid using the keyword variables :)

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+import difflib
+
+import config
+from templates.generic import *
```
Let's import the module that we need explicitly rather than importing everything, very dangerous :)

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+                output['output'] = template.get_message()
+                output['success'] = True
+
+    except:
```
This is unbearable exception, do we know what kind of exception are we catching ?

horrormyth replied:
@swapagarwal  I saw other modules code as well and try and except used seems to be very unbounded and is not helpful.And here @finleylau is trying to make a module here with the same implementation.Would you be able to look at it, please?

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+            search_url += '&cuisines={type}'.format(type=type)
+
+        query = requests.get(search_url,headers=header).json()['restaurants']
+
```
Could you please explain what is happening here :) a bit confusing and may be I sense that it could be simplified from the line 25, And it would be better if we have a separate function with the readable name when the code is longer.

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+
+        query = requests.get(search_url,headers=header).json()['restaurants']
+
+        if name == '':
```
Could you please explain what is happening from here :) a bit confusing and may be I sense that it could be simplified from the line 25, And it would be better if we have a separate function with the readable name when the code is longer.

finleylau replied:
Basically, queries the api again for more specific results on a specific restaurant or type of food from the input

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,93 @@
+        output['error_msg'] = TextTemplate(error_message).get_message()
+        output['success'] = False
+
+    return output
```
Let's add blank line at the end :)

horrormyth commented on modules/tests/test_food.py:
```diff
@@ -0,0 +1,8 @@
+    assert ('food' == modules.process_query('Is Giordano\'s pizza in Chicago good?')[0])
+    assert ('food' == modules.process_query('Chipotle in Evanston')[0])
+    assert ('food' == modules.process_query('What are some restaurants in Chicago')[0])
+    assert ('food' != modules.process_query('something random')[0])
```
Let's add blank line at the end the final one :)

finleylau replied:
I've added blank lines to the ends of the two files but github's just not displaying them in the preview windows but they're there

finleylau commented:
@horrormyth I fixed the variable names, the line formatting, and the blank lines at the end.  I also deleted the .vs folder (some files Visual Studio made that don't affect the code at all).  I'm not sure how to deal with the JSON conversion failures or the exception that you were talking about so I could use a little help there :)

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,123 @@
+				error_message += '\n  - Chipotle in Evanston'
+				error_message += '\n  - What are some restaurants in Chicago?'
+				output['error_msg'] = TextTemplate(error_message).get_message()
+				output['success'] = False
```
@finleylau  This seems to be duplicated everywhere, Could we modularize the logic, please :)

finleylau replied:
done!

horrormyth commented on modules/src/food.py:
```diff
@@ -0,0 +1,123 @@
+							output['input'] = input
+							output['output'] = template.get_message()
+							output['success'] = True
+				except (KeyError,IndexError):
```
@finleylau Do you think we will be getting KeyyErro and Index Error here? Because it seems that you have a chunk of code in the try block. Let's modularize, please :)

finleylau replied:
Can you please further explain?  Sorry, I'm not sure what you mean!
